src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -578,10 +649,63 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
     auto config = selectedPd->getConfig();
     config.outConfs.front().setMemDesc(mem_desc);
-    //bypass any checks, we enforce the child descriptor
+    // bypass any checks, we enforce the child descriptor
     selectedPd->setConfig(config);
 }
 
+void MemoryInput::selectOptimalPrimitiveDescriptor() {
+    MemoryInputBase::selectOptimalPrimitiveDescriptor();
+    if (haveSubgraph()) {
+        auto* prim_desc = getSelectedPrimitiveDescriptor();
```

Comment:
I ever thought `MemoryInputBase::selectOptimalPrimitiveDescriptor` deduced out wrong primitive, I had root-caused and fixed them, and adopted your suggestion code here.
This is because original `MemoryInput` only support 1 input, we have to enable  it to support multiple inputs after introducing SubGraph.